ui/src/tempVars/components/MetaQueryTemplateBuilder.tsx

```diff
@@ -0,0 +1,168 @@
+import React, {PureComponent, ChangeEvent} from 'react'
+import _ from 'lodash'
+
+import {proxy} from 'src/utils/queryUrlGenerator'
+import {ErrorHandling} from 'src/shared/decorators/errors'
+import TemplateMetaQueryPreview from 'src/tempVars/components/TemplateMetaQueryPreview'
+import {parseMetaQuery, isInvalidMetaQuery} from 'src/tempVars/utils/parsing'
+
+import {
+  TemplateBuilderProps,
+  RemoteDataState,
+  TemplateValueType,
+} from 'src/types'
+
+const DEBOUNCE_DELAY = 750
```

Comment:
is there a template variables constants file that this could go into?

Reply:
My attitude towards constants is:

- If they are used in more than one place, extract them to a constants module
- If they are only used in one place, keep them in that same place to minimize the need to bounce around the codebase while editing.

Does that seem reasonable?